beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -64,8 +64,31 @@ func (s *Service) processPendingBlocks(ctx context.Context) error {
 			span.End()
 			continue
 		}
-		s.pendingQueueLock.RUnlock()
 		inPendingQueue := s.seenPendingBlocks[bytesutil.ToBytes32(b.Block.ParentRoot)]
+		s.pendingQueueLock.RUnlock()
+
+		blkRoot, err := stateutil.BlockRoot(b.Block)
+		if err != nil {
+			traceutil.AnnotateError(span, err)
+			span.End()
+			return err
+		}
+		parentIsBad := s.hasBadBlock(bytesutil.ToBytes32(b.Block.ParentRoot))
+		blockIsBad := s.hasBadBlock(blkRoot)
+		// Check if parent is a bad block.
+		if parentIsBad || blockIsBad {
+			// Set block as bad if its parent block is bad too.
+			if parentIsBad {
+				s.setBadBlock(blkRoot)
+			}
+			// Remove block from queue.
+			s.pendingQueueLock.Lock()
+			delete(s.slotToPendingBlocks, slot)
+			delete(s.seenPendingBlocks, blkRoot)
+			s.pendingQueueLock.Unlock()
+			span.End()
```

Comment:
Should the span end here? The line after is `continue`

Reply:
we start a new span for each slot/block in the loop, so it makes sense to end it here. The same is done below